exec/src/main/java/com/optum/sourcehawk/exec/FlattenConfigCommand.java

```diff
@@ -0,0 +1,177 @@
+     */
+    private static FlattenConfigResult executeFlatten(final String configurationFileLocation,
+                                                      final SourcehawkConfiguration sourcehawkConfiguration) {
+        byte[] content;
+        try {
+            content = WRITER_WITH_DEFAULT_PRETTY_PRINTER.writeValueAsBytes(sourcehawkConfiguration);
+        } catch (IOException e) {
+            return handleException(configurationFileLocation, e);
+        }
+
+        return FlattenConfigResult.success(content);
+    }
+
+    /**
+     * Handle exceptions from Serialization
+     *
+     * @param configurationFileLocation the configuration location
+     * @param e                         the exception
+     * @return an error flatten result
+     */
+    private static FlattenConfigResult handleException(final String configurationFileLocation, final IOException e) {
+        val message = String.format("Error flattening sourcehawk configuration at %s with error: %s", configurationFileLocation, e.getMessage());
+        return FlattenConfigResult.error(message);
+    }
+
+    private static final String SOURCEHAWK_FLATTENED_YML = "sourcehawk-flattened.yml";
+
+    /**
+     * Log the result of the flatten which will either output to a file or console based on params
+     *
+     * @param flattenResult      the flatten result
+     * @param repositoryFilePath the repository file path
+     */
+    @SuppressWarnings("squid:S2629")
+    void outputResults(final FlattenConfigResult flattenResult,
+                       final Path repositoryFilePath) {
+        if (repositoryFilePath == null || StringUtils.isBlankOrEmpty(repositoryFilePath.toString())) {
+            handleDryRunOutput(flattenResult);
+        } else {
+            handleFileSystemOutput(flattenResult, repositoryFilePath);
+        }
+    }
+
+    /**
+     * Log the result of the flatten to the file system
+     *
+     * @param flattenResult      the flatten result
+     * @param repositoryFilePath the repository file path
+     */
+    private static void handleFileSystemOutput(final FlattenConfigResult flattenResult, final Path repositoryFilePath) {
+        try {
+            String writerPath = StringUtils.defaultString(repositoryFilePath.toString(), SOURCEHAWK_FLATTENED_YML);
+            LocalRepositoryFileWriter.writer().write(writerPath, flattenResult.getContent());
+            Sourcehawk.CONSOLE_RAW_LOGGER.info(flattenResult.getFormattedMessage());
+            Sourcehawk.CONSOLE_RAW_LOGGER.info("Output to {}", writerPath);
+        } catch (IOException e) {
+            Sourcehawk.CONSOLE_RAW_LOGGER.error("Could not flatten file due to {}", e.getMessage());
+        }
+    }
+
+    /**
+     * Log the result of the flatten to the console
+     *
+     * @param flattenResult the flatten result
+     */
+    private static void handleDryRunOutput(final FlattenConfigResult flattenResult) {
+        if (flattenResult != null && flattenResult.getContent() != null) {
```

Comment:
Is there ever a case when `flattenResult` is actually `null`?

Reply:
Well this is Java so maybe?

Reply:
Why would we ever want to produce a NPE?!